src/realm/array_decimal128.cpp

```diff
@@ -65,10 +152,16 @@ void ArrayDecimal128::move(ArrayDecimal128& dst_arr, size_t ndx)
 {
     size_t elements_to_move = m_size - ndx;
     if (elements_to_move) {
+        if (m_width >= dst_arr.m_width) {
+            dst_arr.upgrade_leaf(m_width);
+        }
+        else {
+            upgrade_leaf(dst_arr.m_width);
```

Comment:
Is it necessary to modify the width here? Wouldn't it be more efficient to copy values to the destination one by one (converting widths on the fly) without modifying the src array?

Reply:
I can make it like that.